From bb8b06debad74d45d703479a64d867d0986b687a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Val=C3=A9ry=20Febvre?= <vfebvre@easter-eggs.com>
Date: Fri, 17 Oct 2025 01:20:31 +0200
Subject: [PATCH] Back to colorthief python package

to avoid Rust and save disk space

modern-colorthief is certainly faster, but uses ~1.8MB of disk space!

The palette computation is only performed once per cover, so speed is
not an essential factor.

Close #839

diff --git a/komikku/models/database/mangas.py b/komikku/models/database/mangas.py
index 47651e2e..483f10cc 100644
--- a/komikku/models/database/mangas.py
+++ b/komikku/models/database/mangas.py
@@ -13,7 +13,7 @@ from pathlib import Path
 import shutil
 import time
 
-from modern_colorthief import get_palette
+from colorthief import ColorThief
 from PIL import Image
 from PIL import ImageFilter
 from PIL import ImageStat
@@ -139,7 +139,7 @@ class Manga:
                 if '@define-color' not in data:
                     return data
 
-        palette = get_palette(cover_path, color_count=2)
+        palette = ColorThief(cover_path).get_palette(color_count=2, quality=1)[:2]
         if len(palette) != 2:
             # Single color image?
             return None